File: app/src/main/java/com/pitchai/flickrrecentphotobrowser/presentation/internal/di/component/MainGridListFragmentComponent.java
package com.pitchai.flickrrecentphotobrowser.presentation.internal.di.component;

import com.pitchai.flickrrecentphotobrowser.presentation.fragments.MainGridFragmentList;
import com.pitchai.flickrrecentphotobrowser.presentation.internal.di.PerFragment;
import com.pitchai.flickrrecentphotobrowser.presentation.internal.di.modules
        .MainGridListFragmentModule;

import dagger.Subcomponent;

/**
 * Created by pitchairajamani on 4/24/17.
 */
@PerFragment
@Subcomponent(
        modules = MainGridListFragmentModule.class
)
public interface MainGridListFragmentComponent {
    MainGridFragmentList inject(MainGridFragmentList mainGridFragmentList);
}

File: app/src/main/java/com/pitchai/flickrrecentphotobrowser/presentation/internal/di/component/MainGridListFragmentComponent.kt
package com.pitchai.flickrrecentphotobrowser.presentation.internal.di.component

import com.pitchai.flickrrecentphotobrowser.presentation.fragments.MainGridFragmentList
import com.pitchai.flickrrecentphotobrowser.presentation.internal.di.PerFragment
import com.pitchai.flickrrecentphotobrowser.presentation.internal.di.modules.MainGridListFragmentModule
import dagger.Subcomponent

/**
 * Created by pitchairajamani on 4/24/17.
 */
@PerFragment
@Subcomponent(modules = [MainGridListFragmentModule::class])
interface MainGridListFragmentComponent {
    fun inject(mainGridFragmentList: MainGridFragmentList?): MainGridFragmentList?
}